transformer/common_data_provider.go

```diff
@@ -37,24 +37,41 @@ const (
 
 var uuidNamespace = uuid.Must(uuid.FromString("971a1103-6b5d-4b60-ab3d-8a339a58c6c8"))
 
-func NewCommonDataProvider(log *logp.Logger, cfg *config.Config) (CommonDataProvider, error) {
-	KubeClient, err := providers.KubernetesProvider{}.GetClient(cfg.KubeConfig, kubernetes.KubeClientOptions{})
+func NewCommonDataProvider(log *logp.Logger, cfg *config.Config) CommonDataProvider {
+	k8sAvailable := true
```

Comment:
Consider implementing `K8sCommonDataProvider` and leaving `CommonDataProvider` k8s agnostic.

Reply:
Done